src/swarm-engine/Swarm/Game/State.hs

```diff
@@ -490,6 +493,20 @@ contentAt (Cosmic subworldName loc) = do
     return (maybe BlankT terrainName terrObj, maybeEnt)
   return $ fromMaybe (BlankT, Nothing) val
 
+
+-- | Perform an action requiring a 'Robots' state component in a
+--   larger context with a 'GameState'.
+zoomRecognizer ::
```

Comment:
@byorgey do you know why this `zoomRecognizer` function doesn't work?  I get `Overlapping instances for Control.Effect.Sum.Member` and `Could not deduce ‘Control.Effect.Sum.Member` errors.

I wanted to replace some boilerplate like `discovery . structureRecognition . foundStructures` with just `foundStructures`.  But it seems to be incompatible with the `%=` operator.

As it happens, if I instead define a `zoomDiscovery` function that dives only one level deep into the `GameState` (i.e. `use discovery` instead of `use $ discovery . structureRecognition`), then the callsites work fine with `%=` (though at the callsite I have to specify `structureRecognition . foundStructures` instead of just `foundStructures`).

Reply:
BTW, this is not urgent---I've found an alternative refactoring that works.